### DISCUSSION TITLE
Dynamic ArgTypes options

### DISCUSSION THREAD

**Discussion by webistomin** [storybookjs/storybook#17423] [Help]
Hi! I ran into a problem. I don't know how to change `ArgTypes` options.
Let's say I have a button that accepts a color prop with values `​​"red"` and `"blue"`.

```js
export default {
  title: 'SomeBtn',
  component: SomeBtn,
  argTypes: {
    color: {
      type: 'select',
      options: [
        'red',
        'blue',
      ],
    },
  },
}
```

I have [an addon](https://www.npmjs.com/package/storybook-addon-themes) that switches themes. So, in one theme, I need the options to be `['red', 'blue']` and in the other, like this `['green', 'yellow']`. I saw there is a hook for changing arguments – `useArgs`, but it did not help me.

Maybe there is some hook like `useArgTypes`s that would allow you to change the array of options?

**Comment by blowsie** [comment]
Related.
https://github.com/storybookjs/storybook/issues/12571
https://github.com/storybookjs/storybook/blob/71d100b037b6d7338bbc1540a72d27c9aa9d6a97/code/core/src/manager-api/root.tsx#L516
